web/src/main/java/org/akaza/openclinica/service/InsightReportServiceImpl.java

```diff
@@ -0,0 +1,138 @@
+package org.akaza.openclinica.service;
+
+import org.json.JSONObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.TimeZone;
+
+@Service( "insightReportService" )
+public class InsightReportServiceImpl implements InsightReportService {
+
+    @Autowired
+    private UserService userService;
+
+    private String fileName;
+    public static final String UNDERSCORE = "_";
+
+    SimpleDateFormat sdf_fileName = new SimpleDateFormat("yyyy-MM-dd'-'HHmmssSSS'Z'");
+    protected final Logger logger = LoggerFactory.getLogger(getClass().getName());
+
+    @Override
+    public String getFileName() {
+        return fileName;
+    }
+
+    @Override
+    public String getFilePath(String label, String reportNames, String filePath) {
+        sdf_fileName.setTimeZone(TimeZone.getTimeZone("GMT"));
+        fileName = label + UNDERSCORE + reportNames +
+                UNDERSCORE + sdf_fileName.format(new Date()) + ".csv";
+        return filePath + File.separator + fileName;
+    }
+
+    @Override
+    public void saveToFile(String content, String fullPath) {
+        FileOutputStream fop = null;
+        File file;
+        try {
+
+            file = new File(fullPath);
+            fop = new FileOutputStream(file);
+
+            // get the content in bytes
+            byte[] contentInBytes = content.getBytes();
+
+            fop.write(contentInBytes);
+            fop.flush();
+            fop.close();
+
+        } catch (IOException e) {
+            logger.error("Error creating file : ", e);
+        } finally {
+            try {
+                if (fop != null) {
+                    fop.close();
+                }
+            } catch (IOException e) {
+                logger.error("Error closing output stream : ", e);
+            }
+        }
+    }
+
+    @Override
+    public String doPost(String api, HashMap<String, String> mapHeaders, String body, HashMap<String, String> queries) {
```

Comment:
this method doesn't need to be in the interface anymore since we are only calling this from within this class. You can make this method private.